covid_graphs/covid_graphs/scripts/server.py

```diff
@@ -70,4 +73,15 @@ def covid19_heatmap_polynomial():
     def covid19_heatmap_exponential():
         return covid19_heatmap_exponential_app.index()
 
-    server.run(host="0.0.0.0", port=8081, debug=True, extra_files=data_dir.glob("*.data"))
+    i = inotify.adapters.InotifyTree(str(data_dir))
+
+    p = Process(target=server.run, kwargs=dict(host="0.0.0.0", port=8081))
+    p.start()
+    while (True):
+        events = list(i.event_gen(yield_nones=False, timeout_s=1))
+        if len(events) != 0:
+            p.terminate()
```

Comment:
Add a print statement, that you are restarting the server based on file changes inside the data directory.